### ISSUE TITLE
Jet Set Radio Future: FPS drops on boost on SteamDeck

### ISSUE THREAD

**Issue by Godsbane** [xemu-project/xemu#995]
### Title

Jet Set Radio Future
https://xemu.app/titles/49470018/#JSRF-Jet-Set-Radio-Future

### Bug Description

Game seems to work perfectly until the boost button is hit, and then the framerate tanks to a few FPS at best and the game becomes unplayable until the effect is over.

### Expected Behavior

When hitting the boost button, the framerate should not drop.

### xemu Version

0.7.18

have tried it in each version since i got my deck, starting with 0.6.something. i don't recall exactly, but it has never worked.

### System Information

OS: SteamOS
SteamDeck

### Additional Context

https://cdn.discordapp.com/attachments/968608420028620820/980001078517047336/20220528_015319.mp4

I'm not certain if this will work. I pulled it from a different user's post in discord, showing the issue. Let me know if you need another video.

Also the same thing happens when certain text enters the screen in THPS4. I will get around to making a proper report for that as well.

**Comment by Knucklesfan** [comment]
This is not just a problem on steam deck, but appears to be a universal problem involving linux in general.

**Comment by ShroomKing** [comment]
Same problem on macOS. Renderer bug?

**Comment by Tenseiren** [comment]
Experienced the same issue running macOS. This doesn't appear to be platform specific.

**Comment by jcdj1996** [comment]
Same issue on Windows 10

**Comment by KickAssDave** [comment]
I wonder if it's a specific graphical effect causing this.
Interesting, someone reported a very similar issue here:
https://github.com/mborgerson/xemu/issues/1058

**Comment by DopamineSource** [comment]
yes i think its that ripple/motion blur effect. Gun Valkyrie uses it on Water too to create caustics i believe and about the same amount of slowdown that happens when boosting forward also happens anytime water is shown. 
JSFR also has a similar blurring/rippling effect when boosting and i wouldnt be surprised if that "certain text" in thps4 is a similar thing.

**Comment by OutRunner** [comment]
I think it's the same issue in Ninja Gaiden Black as well (https://github.com/mborgerson/xemu/issues/453). Certain transparency effects cause frame rate drops.

**Comment by HappehLemons** [comment]
I wish we could disable the  synchronization event that was added to improve accuracy which causes this bug (#453). This has been an issue since 2021, so it would be nice to have a workaround to make these game playable again. Would really love to be able to play Ninja Gaiden Black on the steam deck.

**Comment by KerrWasHere** [comment]
Is there a chance of this being addressed anytime soon?

**Comment by Godsbane** [comment]
Zink helps!

**Comment by HappehLemons** [comment]
Maybe @mborgerson can provide some insight as I know this was an issue he was hoping to resolve last year.

**Comment by KreeseXD** [comment]
I remember putting off playing JSRF because of this slowdown; I decided to test it today because I have not tested any of the builds in the past few months and I did not run into this slowdown. Not sure if there was a PR that fixed this that I was not aware or if it's just something with my system, but thought I'd provide an update for those still interested

https://user-images.githubusercontent.com/57853013/210099522-39170dcc-3c88-4c09-b7ca-1babf9edf63f.mp4

XEMU Version: 0.7.71
System: Pop OS 22.04
Ryzen 5 5600x
Radeon 6700 XT

**Comment by Godsbane** [comment]
Same experience on the Deck now. It drops the internal frame rate from 60 to about 55fps, but it's completely playable now.

Should close this issue as completed?

**Comment by jcdj1996** [comment]
Issue still present for me on 0.7.71. Framerate stays stable but 'MSPF' jumps from 4 to 40+
 i7-7700K CPU @ 4.20GHz
Windows 10
 NVIDIA GeForce RTX 3060
Driver:       4.0.0 NVIDIA 527.56
Shader:       4.00 NVIDIA via Cg compiler

**Comment by KerrWasHere** [comment]
I’ll check this out shortly on the deck On Dec 30, 2022, at 1:35 PM, James Jennings ***@***.***> wrote:﻿
Issue still present for me on 0.7.71
i7-7700K CPU @ 4.20GHz
Windows 10
NVIDIA GeForce RTX 3060
Driver:       4.0.0 NVIDIA 527.56
Shader:       4.00 NVIDIA via Cg compiler

**Comment by OutRunner** [comment]
JSRF is OK for me on Steam Deck too. Ninja Gaiden Black still has the heavy slowdowns so I guess it’s a different issue.

**Comment by ordinaryashen** [comment]
YAY! Gonna try this now also, such awesome news!

**Comment by HappehLemons** [comment]
Hey @OutRunner, do you know if the Ninja Gaiden Black still is having slow down caused by this regression?

**Comment by OutRunner** [comment]
@HappehLemons yeah it does seem so. The moves where Ryu leaves behind transparent echoes of himself and the times when dead enemies “dissolve” bloodily still have a lot of slowdown on Steam Deck. I use the version @pas-de-2 kindly built and uploaded here: https://github.com/xemu-project/xemu/issues/837#issuecomment-1435993325 for Ninja Gaiden and its mostly full speed.

**Comment by HappehLemons** [comment]
Oh, I didn't know this existed. I'll have to give this a try, thanks @OutRunner!

**Comment by cheatfreak47** [comment]
Yeah this still seems to be happening on some non-steam deck platforms, can confirm I am seeing tremendous FPS drops on an i7  7700 rig as well.

**Comment by KreeseXD** [comment]
After being away from testing for years (https://github.com/xemu-project/xemu/issues/995#issuecomment-1368034051), I decided to recheck this issue out of curiosity since I'm on a new rig. I still don't get this issue

Specs:
OS: Windows 11
CPU: 13th Gen Intel(R) Core(TM) i9-13980HX
GPU: NVIDIA GeForce RTX 4080 Laptop GPU/PCIe/SSE2 24H2
Renderer: OpenGL @ 1x scaling
XEMU version: 0.8.35


https://github.com/user-attachments/assets/7390f951-d7e0-4d7e-8c69-34a370426a56

**Comment by CODEX0827** [comment]
for me the slowdown happens when I up the resolution from native to something else

**Comment by cheatfreak47** [comment]
Can confirm this issue seems present still on the Steam Deck as well (got one a few months ago)

 Overall, I think this issue thread is more of "please optimize this specifically because JSRF should be a reference game for Xemu" rather than "oh this doesn't work / is broken for everyone" type issue.

Clearly if you have a mega beefy CPU this is a non-issue, based on some of the clips posted.

**Comment by KerrWasHere** [comment]
Agreed, but the steam deck should be able to run this with no issues either wayOn Aug 16, 2025, at 12:35 PM, CheatFreak ***@***.***> wrote:﻿cheatfreak47 left a comment (xemu-project/xemu#995)
Can confirm this issue seems present still on the Steam Deck as well (got one a few months ago)
Overall, I think this issue thread is more of "please optimize this specifically because JSRF should be a reference game for Xemu" rather than "oh this doesn't work / is broken for everyone" type issue.
Clearly if you have a mega beefy CPU this is a non-issue, based on some of the clips posted.
